components/tidb_query/src/codec/mysql/time/mod.rs

```diff
@@ -57,605 +48,1163 @@ const MONTH_NAMES_ABBR: &[&str] = &[
     "Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec",
 ];
 
-#[inline]
-fn zero_time(tz: &Tz) -> DateTime<Tz> {
-    tz.timestamp(ZERO_TIMESTAMP, 0)
+fn is_leap_year(year: u32) -> bool {
+    year & 3 == 0 && (year % 100 != 0 || year % 400 == 0)
 }
 
-#[inline]
-pub fn zero_datetime(tz: &Tz) -> Time {
-    Time::new(zero_time(tz), TimeType::DateTime, mysql::DEFAULT_FSP).unwrap()
+fn last_day_of_month(year: u32, month: u32) -> u32 {
+    match month {
+        4 | 6 | 9 | 11 => 30,
+        2 => is_leap_year(year) as u32 + 28,
```

Comment:
Would this be better: 

```rust
2 if is_leap_year(year) => 29,
2 => 28
```

Reply:
With this kind of code, the compiler could apply a conditional move here.